assets/app/scripts/services/data.js

```diff
@@ -360,6 +360,124 @@ angular.module('openshiftConsole')
     return deferred.promise;
   };
 
+// https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/btoa
+function utf8_to_b64( str ) {
+    return window.btoa(window.unescape(encodeURIComponent( str )));
+}
+function b64_to_utf8( str ) {
+    return decodeURIComponent(window.escape(window.atob( str )));
+}
+
+// TODO (bpeterse): ideally DataService would be more general & log specific function
+// could be in a separate service, but we need the helper functions
+DataService.prototype.createStream = function(kind, name, context, isRaw) {
+  var getNamespace = this._getNamespace.bind(this);
+  var urlForResource = this._urlForResource.bind(this);
+  kind = this.kindToResource(kind) ?
+              this.kindToResource(kind) :
+              normalizeResource(kind);
+
+  var protocols = isRaw ? 'binary.k8s.io' : 'base64.binary.k8s.io';
+  var identifier = 'stream_';
+  var openQueue = {};
+  var messageQueue = {};
+  var closeQueue = {};
+  var errorQueue = {};
+
+  var stream;
+  var makeStream = function() {
+     return getNamespace(kind, context, {})
+                .then(function(params) {
+                  return  $ws({
+                            url: urlForResource(kind, name, null, context, true, _.extend(params, {follow: true})),
+                            onopen: function(evt) {
+                              _.each(openQueue, function(fn) {
+                                fn(evt);
+                              });
+                            },
+                            onmessage: function(evt) {
+                              if(!_.isString(evt.data)) {
+                                Logger.log('log stream response is not a string', event.data);
```

Comment:
`evt` is short for `event` but longer than `e`.  Perhaps is more confusing this way?  :frowning: